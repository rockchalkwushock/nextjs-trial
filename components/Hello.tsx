import * as React from 'react'
import styled from 'styled-components'

interface Props {
  framework: string
}

const Title = styled.h1`
  font-size: 2rem;
  color: magenta;
`

export const Hello: React.SFC<Props> = ({ framework }) => (
  <Title>Hello from {framework}!</Title>
)
