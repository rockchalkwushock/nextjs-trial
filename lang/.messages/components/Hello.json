[
  {
    "id": "greeting",
    "defaultMessage": "Hello from NextJS"
  }
]