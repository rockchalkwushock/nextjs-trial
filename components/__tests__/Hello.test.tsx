import * as React from 'react'
import { render } from 'react-testing-library'

import { Hello } from '../Hello'

describe('<Hello />', () => {
  test('should render to DOM', () => {
    const { asFragment, container } = render(<Hello framework="NextJS" />)
    expect(container.textContent).toEqual('Hello from NextJS!')
    expect(asFragment()).toMatchSnapshot()
  })
})
