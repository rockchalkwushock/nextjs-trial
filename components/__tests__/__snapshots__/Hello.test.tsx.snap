// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Hello /> should render to DOM 1`] = `
<DocumentFragment>
  <h1
    class="Hello__Title-dz5c7s-0 kCTxqR"
  >
    Hello from NextJS!
  </h1>
</DocumentFragment>
`;
